contrib/scalar/scalar.c

```diff
@@ -21,24 +21,28 @@ static int is_unattended(void) {
 
 /**
  * Remove the deepest subdirectory in the provided path string. Path must not
- * include a trailing path separator.
+ * include a trailing path separator. Returns 1 if parent directory found,
+ * otherwise 0.
```

Comment:
By the same logic as I discussed regarding `find_enlistment()`, I would consider `strbuf_parentdir()` _not_ to be a "yes/no" question, i.e. return `0` on success and `-1` if there is no parent directory.

Of course, `strbuf_parentdir()` is a slightly more mucky thing than `find_enlistment()` because we actually do not necessarily expect a parent directory to exist, so... I _could_ be convinced to leave this as a Boolean ;-)

Reply:
My thought on this is that `strbuf_parentdir` not finding the parent isn't really an "error", just an indicator that you're at the root (similar to running `cd ..` at the root directory). It's also not *completely* unprecedented (although there also aren't a ton of examples like it either), as `strbuf_strip_suffix` behaves similarly (returning a boolean indicating whether the operation could be performed as expected or not). And, since it's

Reply:
Valid point. Color me convinced ;-)